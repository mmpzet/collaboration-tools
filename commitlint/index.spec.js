/* eslint-disable max-nested-callbacks */
const load = require('@commitlint/load');
const lint = require('@commitlint/lint');

describe('commitlint configuration', () => {
  it('should be a valid configuration', () => load({
    // commitlint extends path is always relative to project root
    extends: ['./commitlint']
  }));

  describe('loaded configuration', () => {
    let config;

    beforeAll(async () => {
      config = await load({
        // commitlint extends path is always relative to project root
        extends: ['./commitlint']
      });
    });

    it('should identify INVALID commit message', async () => {
      const invalidMessages = [
        'missing type',
        'invalid: invalid type',
        'feat:not spaced',
        'feat:',
        'feat: some very very very very very very very very very very very very very very very long text'
      ];

      await Promise.all(invalidMessages.map(async (message) => {
        const report = await lint(message, config.rules);

        report.message = message;

        // console.log(message, report)
        expect(report.valid).toBe(false);
        expect(report).toMatchSnapshot();
      }));
    });

    it('should identify VALID commit message', async () => {
      const validMesages = [
        'feat: some valid text'
      ];

      await Promise.all(validMesages.map(async (message) => {
        const report = await lint(message, config.rules);

        report.message = message;
        expect(report.valid).toBe(true);
        expect(report).toMatchSnapshot();
      }));
    });
  });
});
