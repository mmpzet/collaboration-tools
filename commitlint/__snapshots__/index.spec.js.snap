// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commitlint configuration loaded configuration should identify INVALID commit message 1`] = `
Object {
  "errors": Array [
    Object {
      "level": 2,
      "message": "subject may not be empty",
      "name": "subject-empty",
      "valid": false,
    },
    Object {
      "level": 2,
      "message": "type may not be empty",
      "name": "type-empty",
      "valid": false,
    },
  ],
  "input": "missing type",
  "message": "missing type",
  "valid": false,
  "warnings": Array [],
}
`;

exports[`commitlint configuration loaded configuration should identify INVALID commit message 2`] = `
Object {
  "errors": Array [
    Object {
      "level": 2,
      "message": "type must be one of [feat, fix, docs, style, refactor, perf, test, chore]",
      "name": "type-enum",
      "valid": false,
    },
  ],
  "input": "invalid: invalid type",
  "message": "invalid: invalid type",
  "valid": false,
  "warnings": Array [],
}
`;

exports[`commitlint configuration loaded configuration should identify INVALID commit message 3`] = `
Object {
  "errors": Array [
    Object {
      "level": 2,
      "message": "subject may not be empty",
      "name": "subject-empty",
      "valid": false,
    },
    Object {
      "level": 2,
      "message": "type may not be empty",
      "name": "type-empty",
      "valid": false,
    },
  ],
  "input": "feat:not spaced",
  "message": "feat:not spaced",
  "valid": false,
  "warnings": Array [],
}
`;

exports[`commitlint configuration loaded configuration should identify INVALID commit message 4`] = `
Object {
  "errors": Array [
    Object {
      "level": 2,
      "message": "subject may not be empty",
      "name": "subject-empty",
      "valid": false,
    },
    Object {
      "level": 2,
      "message": "type may not be empty",
      "name": "type-empty",
      "valid": false,
    },
  ],
  "input": "feat:",
  "message": "feat:",
  "valid": false,
  "warnings": Array [
    Object {
      "level": 1,
      "message": "header must not be shorter than 10 characters, current length is 5",
      "name": "header-min-length",
      "valid": false,
    },
  ],
}
`;

exports[`commitlint configuration loaded configuration should identify INVALID commit message 5`] = `
Object {
  "errors": Array [
    Object {
      "level": 2,
      "message": "header must not be longer than 50 characters, current length is 95",
      "name": "header-max-length",
      "valid": false,
    },
  ],
  "input": "feat: some very very very very very very very very very very very very very very very long text",
  "message": "feat: some very very very very very very very very very very very very very very very long text",
  "valid": false,
  "warnings": Array [],
}
`;

exports[`commitlint configuration loaded configuration should identify VALID commit message 1`] = `
Object {
  "errors": Array [],
  "input": "feat: some valid text",
  "message": "feat: some valid text",
  "valid": true,
  "warnings": Array [],
}
`;
